### PR TITLE
feat: `verdi process status --max-depth`

### LINKED ISSUE #5557: Allow `verdi process list` to accept more than one group when filtering with `-G`
When doing `verdi process list` one can specify to only list the processes from a group by providing it with the `-G` option, but the command only seems to accept one group like this. It would be nice to be able to provide more than one group for the filtering.

### PR DESCRIPTION
**Pull request by ltalirz**
AiiDA workflows can get very complex with many layers, leading `verdi
process status` to print hundreds of lines of text.

The `--max-depth` allows the user to "zoom out" and look at the bigger
picture, while still being able to "zoom in" on subworkflows of interest
by calling `verdi process status` on those.

Example
```
$ verdi process status --max-depth 2 17460
VerificationWorkChain<17460> Waiting [4:if_(is_verify_convergence)(1:run_convergence)]
    ├── parse_pseudo_info<17461> Finished [0]
    ├── DeltaMeasureWorkChain<17464> Finished [0] [6:finalize]
    ├── _CachingConvergenceWorkChain<18301> Finished [0] [10:finalize]
    └── ConvergenceCohesiveEnergyWorkChain<18577> Waiting [6:run_reference]
```

Fixes #5557

### CODE REVIEW

**Comment by ltalirz** [comment]
@sphuber feel free to chime in as well.

This PR is made against support/1.6.x because the AiiDa SSSP workflows aren't migrated to AiiDA 2.0 yet.
Updating this PR for 2.0 should be easy though, @unkcpz can you take care of this after this PR is merged?

File: aiida/cmdline/utils/ascii_vis.py [hunk 1]
```diff
@@ -279,25 +279,37 @@ def calc_info(node):
...
+    if max_depth == 0:
+        return ''
+
     info_string = info_fn(calc_node)
     called = calc_node.called
     called.sort(key=lambda x: x.ctime)
-    if called:
-        return info_string, [build_call_graph(child, info_fn) for child in called]
+    if called and max_depth != 1:
+        return info_string, [build_call_graph(child, max_depth=max_depth - 1, info_fn=info_fn) for child in called]
```
Comment (unkcpz) [inline aiida/cmdline/utils/ascii_vis.py:312]: If `max_depth` is `-1`, then the function will be then called with `-2`, `-3` ... lead to an infinity iteration?

**Comment by unkcpz** [comment]
> This PR is made against support/1.6.x because the AiiDa SSSP workflows aren't migrated to AiiDA 2.0 yet.

Thanks a lot @ltalirz. I personally do not need this feature since as you know I got a tricky way to inspect my workflow. I think this can directly go to AiiDA 2.0. Migrating my workflow to 2.0 will be the next highest priority after I have calculations in the queue are finished.

File: aiida/cmdline/utils/ascii_vis.py (hunk 1 above)
Comment (ltalirz) [reply aiida/cmdline/utils/ascii_vis.py:312]: No, if `max_depth` is `-1`, then the stopping condition is given by the finiteness of the workflow as before.

Comment (unkcpz) [reply aiida/cmdline/utils/ascii_vis.py:312]: Where is this `-1` condition handled? there are no other functions called.

**Comment by unkcpz** [comment]
> Just to clarify, _I_ needed the feature, and here is the commit to add it to 1.6.x, so I think we should merge this and then add it to `main` as well.

Sure! I'll take care of this after this PR is merged?

File: aiida/cmdline/utils/ascii_vis.py (hunk 1 above)
Comment (ltalirz) [reply aiida/cmdline/utils/ascii_vis.py:312]: The `-1` condition does not need to be handled.
The stopping condition is implicit in the finiteness of the call graph (`called` will eventually be empty).
Just look at the code before this commit.

Comment (ltalirz) [reply aiida/cmdline/utils/ascii_vis.py:312]: Sorry, I see what you mean ;-)

Comment (ltalirz) [reply aiida/cmdline/utils/ascii_vis.py:312]: I'll add a test for `-1` as well

Comment (ltalirz) [reply aiida/cmdline/utils/ascii_vis.py:312]: ah wait, max depth `-1` should actually print everything, so I confused myself ;-)

I will add a test with `0` to prove that the output is empty.

Comment (unkcpz) [reply aiida/cmdline/utils/ascii_vis.py:312]: I see your point, I think it will be more clear if the `elif` used here to exclusively separate all conditions?
```
if max_depth is None or max_depth == -1:
    return info_string
elif max_depth == 0:
    return ''
else:
  # build_call_graph()
```

Comment (ltalirz) [reply aiida/cmdline/utils/ascii_vis.py:312]: Hi @unkcpz, your proposed code is not correct.
If `max_depth` is `None` (or `-1`), the whole function body needs to be evaluated.

It would be possible to deal with positive values of `max_depth` + `None` as well, one would then just need to introduce a special case in the recursive call (i.e. reduce `max_depth` is only reduced if it is not `None`).

I personally consider the code clean as it is, but I'll change it and then you can tell me if it is less confusing to you.

Comment (ltalirz) [reply aiida/cmdline/utils/ascii_vis.py]: I've made the change. Not sure whether the code is more elegant this way, but there you go

Comment (unkcpz) [reply aiida/cmdline/utils/ascii_vis.py]: Thanks a lot @ltalirz. It is more clear to me now.

Comment (unkcpz) [reply aiida/cmdline/utils/ascii_vis.py]: Wait, I look at you previous version where there is one layer of `if`.

Comment (unkcpz) [reply aiida/cmdline/utils/ascii_vis.py]: But all good. I totally understand the logic now.

Comment (unkcpz) [reply aiida/cmdline/utils/ascii_vis.py]: It is all very clean after catch `raise ValueError('max_depth must be >= 0')`.

**Review by unkcpz — APPROVED** [review]

**Comment by ltalirz** [comment]
The verdi load time check is failing but I don't think this PR has the potential to increase load time. Will merge.
